saleor/order/utils.py

```diff
@@ -343,10 +347,67 @@ def match_orders_with_new_user(user: User) -> None:
     Order.objects.confirmed().filter(user_email=user.email, user=None).update(user=user)
 
 
+def chunk_products(products, product_limit):
+    """Split products to list of chunks.
+
+    Each chunk represents products per page, product_limit defines chunk size.
+    """
+    chunks = []
+    for i in range(0, len(products), product_limit):
+        limit = i + product_limit
+        chunks.append(products[i:limit])
+    return chunks
+
+
+def get_product_limit_first_page(products):
+    MAX_PRODUCTS_WITH_TABLE = 3
+    MAX_PRODUCTS_WITHOUT_TABLE = 4
+
+    if len(products) < MAX_PRODUCTS_WITHOUT_TABLE:
+        return MAX_PRODUCTS_WITH_TABLE
+
+    return MAX_PRODUCTS_WITHOUT_TABLE
+
+
 def generate_invoice_pdf_for_order(invoice):
-    logo_path = static_finders.find("images/logo-light.svg")
+    logo_path = static_finders.find("images/logo.svg")
+    MAX_PRODUCTS_PER_PAGE = 13
+
+    if not getattr(invoice, "order"):
```

Comment:
This validation should be part of mutation instead of part of the utils. 
When using som validation in utils we shouldn't use "GraphQL errors".

Reply:
Agree, `ValidationError`s should ideally live in the GraphQL layer.

Reply:
Yeah, furthermore I think I might have covered that already, will remove it from here.